### DISCUSSION TITLE
Access internet even in the middle of protests

### DISCUSSION THREAD

**Discussion by norsnap** [WeAreMahsaAmini/Core#62] [Ideas]
The only protocol that works in Iran at all hours of the day, even with cellular connection in the middle of protests, is **shadowsocks+v2ray+websocket+tls+cdn**.

It doesn't need a VPS inside Iran, this method hides your VPS IP address behind an IP address from a major CDN provider like Arvancloud, and most ISPs do not block those IP addresses. So Arvancloud's CDN can be used to bypass Internet outages.

1- Buy simple server (VPS) for 5 dollars outside of Iran.
2- Buy cheap domain for 2 dollars.
3- Create [Arvancloud](https://www.arvancloud.com/) account. It only needs an email address and a phone number (both Iranian and international phone numbers accepted)
4- Login to Arvancloud account and connect your domain to your server's IP address.
5- From the "DNS records" section of the Arvancloud panel, activate the "Cloud Service" option.
6- From the "HTTPS settings" section, select the issue certificate.
7- From the "HTTPS settings" section, activate the "Activate HTTPS" option.
8- On the VPS, start shadowsocks+v2ray service with port 443, and we can use the following docker files. (two errors will occur during startup that can be fixed. Use docker compose logs to see the errors. I will fork this repo, fix the errors)

https://github.com/haohaolee/shadowsocks-v2ray-tls-ws

Clients for Windows, Linux and Mac can be obtained from the GitHub page of the project.
https://github.com/shadowsocks

For IOS, we need the following app, which unfortunately is paid, but with tools like AltServer, we can side load the app for free.
https://apps.apple.com/us/app/shadowrocket/id932747118
https://www.iphonecake.com/app_932747118_.html

For Android, we need to install the following two apps.
https://play.google.com/store/apps/details?id=com.github.shadowsocks
https://play.google.com/store/apps/details?id=com.github.shadowsocks.plugin.v2ray

Friends who have technical ability, please create this service and share it with your friends and family.

**Comment by anonymoper** [comment]
Hey, check this related idea
 [https://github.com/WeAreMahsaAmini/Core/discussions/66](https://github.com/WeAreMahsaAmini/Core/discussions/66)

**Comment by AnonymousMaximus2** [comment]
to start this idea and grow it and widespread it, you need funding resource and for funding you need to establish an international company for example in USA,

please check explanation in below discussion:


https://github.com/WeAreMahsaAmini/Core/discussions/115

**Comment by norsnap** [comment]
Can we get a repo for this to a share detailed guide, to aggregate config files provided by technical users and further improve the development?
@syavash @farnoodma

**Reply by syavash** [reply]
@aboutsajjad For this one, I suggest adding a new Markdown file to the Core repo and link the main page to it (if it is only sharing information).

It'll receive way more attention. I'm totally open to creating a new repo, but wanted to offer another option.

**Reply by norsnap** [reply]
@syavash The repo will contain all the materials (docker files,...) needed to spin up the server, and  maybe a web ui to automate server setup for non technical users who wants to participate, and of course a link will be added to the main page for how-to-connect and sharing client configs.

**Reply by IranZop** [reply]
if we combine this work with #249 , we can automate the whole process for major cloud service providers such as Azure and AWS. basically the docker image defined [here](https://github.com/haohaolee/shadowsocks-v2ray-tls-ws) can be used in AWS EC2 boostrap logic to run the docker container

**Reply by norsnap** [reply]
@IranZop sounds amazing.
@syavash can we get that repo?

**Reply by 0xland** [reply]
@syavash @farnoodma any update!?

**Reply by farnoodma** [reply]
Hey everyone, I created the repo [here](https://github.com/WeAreMahsaAmini/FreeInternet). I will update the contribution guide, a channel in Discord + a project board in Trello. Will update you on it.
For starters, we need someone to structure the repo, I added @mona-lisa-1234 as the maintainer. Feel free to start working on it.

**Comment by iliyofksy** [comment]
https://github.com/haohaolee/shadowsocks-v2ray-tls-ws

**Reply by norsnap** [reply]
sorry typo, link fixed
https://github.com/haohaolee/shadowsocks-v2ray-tls-ws

**Comment by AnonymousMaximus2** [comment]
Also please check this

https://github.com/WeAreMahsaAmini/Core/discussions/290

**Comment by thisisnobodyx** [comment]
I have a Volume licensing plan with VPN Express that is working very well in Iran at the moment. Let me know if i can help.

**Comment by AnonymousMaximus2** [comment]
Psiphon 3 is working in Iran, 

does anyone knows what kind of technology it uses?

**Comment by farnoodma** [comment]
As I mentioned here, this idea has been kicked of: https://github.com/WeAreMahsaAmini/Core/discussions/62#discussioncomment-3774382

**Comment by hiddify-com** [comment]
Is it a safe method?
can they detect that we are using arvancloud as proxy?
Because i've seen that all proxy servers are detected and blocked

**Comment by akbartala** [comment]
For anyone considering running the docker files there are two errors on Ubuntu 20.04, which can be solved as below:

### Solving the problem with acme.sh
In `shadowsocks-v2ray-tls-ws/build/start.sh` add the following line after the line starting with `RENEW_SKIP=2`:

```bash
acme.sh  --register-account  -m mypersonalsecureemail@proton.me --server zerossl
```
> Exaplanation: acme.sh requires a one time registration which will be taken care of by adding the line before issuing the certificate

### Solving the problem with ss-server exiting:
In `/root/shadowsocks-v2ray-tls-ws/scripts/start-server.sh` replace the `$SERVER_ADDR` variable with `0.0.0.0` 

> Explanation: The variable is not filled anywhere. Fill the variable or just replace it with `0.0.0.0` as shown.

### Running behind the CDN
 - Lower the minimum TLS version if you get error
 - Make sure that the origin communication protocol is set to HTTPS (WS(S) is sent over HTTPS).
 - Before activating the the CDN proxy, run the docker-compose up once (there are some errors when issuing a certificate with acme.sh over active CDN. Then you can safely activate the CDN proxy.

> You can also run the setup without any CDN by making an NGINX proxy from your local server to your origin server.  

 ### Important hint for configuring the client:
 The websocket path that activates the shadowsocks server is by default set to `/v2ray` 
 
### How does a correct setup look like?
Use the following site: https://www.piesocket.com/websocket-tester to test the websocket server `wss://yourdomain.com/v2ray`

**Reply by 0xland** [reply]
Nice debugging, Can you also pull request it to the repo?

**Comment by matinirani** [comment]
آیا این امکان وجود داره که وقتی اینترانت رو به اینترنت تبدیل کرد با این روش؟ مشکل اساسی در ایران نبود کانکشن هست، زمانی که کانکشن هست مردم بالاخره فیلتر شکن رو پیدا میکنن.

**Comment by matinirani** [comment]
موضوع بعدی جمع آوری پول برای خرید دیش استارلینک هست. که ندیدم همچین موضوعی ؟ کسی هست که تمایل به همکاری داشته باشه.

**Reply by theStratus** [reply]
چجوری میخوای ببریشون داخل ایران؟ 
و اینکه ایا هنوز بستر فنیش از طرف استارلینک اماده است مگه؟

**Comment by politicalcitizen** [comment]
I'm working on the following idea and I appreciate any input on it:
1- Instead of trying to build our own solutions from scratch, we can rely on VPN solutions that already exist and work OK in Iran, like Express VPN or Proton VPN. The problem is that only the paid version of these services have an acceptable quality.
2- We can negotiate with these companies on deals for bulk purchase of VPN accounts, for reduced prices.
3- We can then distribute these free high quality VPN accounts among thousands of people in Iran, with some smart targeting, for example, students or journalists, because they're instrumental in disseminating information publicly.
4- As for funding and covering the costs, we could register a non-profit and gather donations from Iranians living abroad.

This way we could provide VPN service for 10 people for as little as $2 a month, if not less, and that would be without the burden of developing or maintaining the technical solution ourselves. So for something like $2000 a month we should be able to easily provide access to 10,000 people, if not more. And by smartly choosing our target group for receiving these accounts we could maximize the impact.

Another advantage of outsourcing the technical problem to these companies and paying them instead, is that they have the financial incentive to keep their product updated and well-maintained in order to get paid. For example, Proton VPN has already shown its interest by developing Stealth protocol that is design to work in countries with high levels of Internet censorship.

Again I appreciate you letting me know what you think about this idea, especially @AnonymousMaximus2 given your experience with starting organizations.

**Comment by nima7894** [comment]
دوستان من میخوام با استفاده از v2ray+tls+cdn+website روی ابرآروان پروکسیمو راه اندازی کنم. قبل از فعال کردن ابرآروان ( آبی شدن نماد ابر ) پروکسی کار میکنه ولی بعد از فعال شدن ابر دیگه پروکس وصل نمیشه. همچنین با فعال بودن ابر میتونم به وبسایت سرورم با https وصل شم. در ضمن این روشو قبلا روی کلودفلر زده بودم و کار میکرد ولی این اواخر به علت اختلال روی کلودفلر دیگه به کیفیت قبل نبود.

**Reply by navidkhalilian** [reply]
مشکل از ابر آروان هست. باید certificate دو طرف یعنی سمت سرور شما و روی dns یکی باشه

**Comment by ghost** [comment]
#مرگ_بر_آشوبگر

**Comment by rezaro3** [comment]
چرا این همه سر درد؟ از ssh استفاده کنید با بیت وایز یک پروکسی میسازه تمام

**Comment by hamedsbt** [comment]
This thread has many interesting contents: https://github.com/net4people/bbs/issues
